tests/diffpy/utils/scattering_objects/test_diffraction_objects.py

```diff
@@ -232,25 +232,52 @@ def test_diffraction_objects_equality(inputs1, inputs2, expected):
 
 
 def test_q_to_tth():
-    # valid q values including edge cases when q=0 or tth=180 after converting
-    # expected tth values are 2*arcsin(q)
+    # Valid q values that should result in 0-180 tth values after conversion
+    # expected tth values are 2*arcsin(q) in degrees
     actual = DiffractionObject(wavelength=4 * np.pi)
     setattr(actual, "on_q", [[0, 0.2, 0.4, 0.6, 0.8, 1], [1, 2, 3, 4, 5, 6]])
     actual_tth = actual.q_to_tth()
     expected_tth = [0, 23.07392, 47.15636, 73.73980, 106.26020, 180]
     assert np.allclose(actual_tth, expected_tth)
 
 
-def test_q_to_tth_bad():
-    # invalid wavelength or q values when arcsin value is not in the range of [-1, 1]
-    actual = DiffractionObject(wavelength=4 * np.pi)
-    setattr(actual, "on_q", [[0.6, 0.8, 1, 1.2], [1, 2, 3, 4]])
+params_q_to_tth_bad = [
+    # UC1: user did not specify wavelength
+    (
+        [None, [0, 0.2, 0.4, 0.6, 0.8, 1]],
+        "Wavelength is not specified. Please provide a valid wavelength, "
+        "e.g., DiffractionObject(wavelength=0.71).",
+    ),
+    # UC2: user specified invalid q values that result in tth > 180 degrees
+    (
+        [4 * np.pi, [0.2, 0.4, 0.6, 0.8, 1, 1.2]],
+        "Wavelength * q > 4 * pi. Please check if you entered an incorrect wavelength or q value.",
```

Comment:
I think this behavior is good.  The first sentence is a bit mathematical and cryptic. Is there a more "chemist friendly" way of saying it?

Reply:
Does "The combination of wavelength and q values is too large" sound good?

Reply:
How about: "the supplied q-array and wavelength will result in an impossible two-theta.  Please check these values and re-instantiate the DiffractionObject"